pkg/apis/vsphere/v1/template_test.go

```diff
@@ -0,0 +1,112 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+	"go.anx.io/go-anxcloud/pkg/client"
+)
+
+var _ = Describe("Template API bindings", func() {
+	Context("BuildNumber method", func() {
+		It("can parse valid Build identifier", func() {
+			t := Template{Build: "b04"}
+			Expect(t.BuildNumber()).To(Equal(4))
+		})
+
+		DescribeTable("invalid build identifier", func(build string) {
+			t := Template{Build: build}
+			_, err := t.BuildNumber()
+			Expect(err).To(MatchError(ErrFailedToParseTemplateBuildNumber))
+		},
+			Entry("with empty build string", ""),
+			Entry("without build digits", "b"),
+			Entry("with unknown build prefix", "c123"),
+			Entry("with characters between build digits", "b1c23"),
+		)
+	})
+
+	Context("with mocked server", func() {
+		var a api.API
+		location := corev1.Location{Identifier: "mock-location-id"}
+
+		BeforeEach(func() {
+			srv := ghttp.NewServer()
+			srv.AppendHandlers(ghttp.CombineHandlers(
+				ghttp.VerifyRequest("GET", fmt.Sprintf("/api/vsphere/v1/provisioning/templates.json/%s/templates", location.Identifier)),
+				ghttp.RespondWithJSONEncoded(200, mockedTemplateList()),
+			))
+
+			var err error
+			a, err = api.NewAPI(api.WithClientOptions(
+				client.BaseURL(srv.URL()),
+				client.IgnoreMissingToken(),
+			))
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("can List templates", func() {
+			var channel types.ObjectChannel
+			err := a.List(context.TODO(), &Template{Type: TypeTemplate, Location: location}, api.ObjectChannel(&channel))
+			Expect(err).ToNot(HaveOccurred())
+
+			templateCount := 0
+			for res := range channel {
+				t := Template{}
+				err := res(&t)
+				Expect(err).NotTo(HaveOccurred())
+				templateCount++
+			}
+			Expect(templateCount).To(Equal(10))
+		})
+
+		It("can Get templates", func() {
+			tpl := Template{Identifier: "26a47eee-dc9a-4eea-b67a-8fb1baa2fcc0", Type: TypeTemplate, Location: location}
+			err := a.Get(context.TODO(), &tpl)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(tpl.Name).To(Equal("Flatcar Linux Stable"))
+		})
+
+		DescribeTable("find named template", func(name, build, expectedID string) {
+			template, err := FindNamedTemplate(context.TODO(), a, name, build, location)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(template.Identifier).To(Equal(expectedID))
+		},
+			Entry("latest with empty build", "Debian 11", "", "ec547552-d453-42e6-987d-51abe703c439"),
```

Comment:
with `BuildNumber()` returning an error and `FindNamedTemplate()` also returning that error, this will, of course, not succeed anymore and needs to be adapted.

Reply:
This `Entry` does not create a `Template` on which `BuildNumber` is called. The empty build string is an alias for `LatestTemplateBuild` in `FindNamedTemplate`.

Reply:
uhm.. `BuildNumber()` is called whenever we want the latest template, so in this case it is called?

But since we decided to log a warning in `FindNamedTemplate` and not error out, this does not produce an error and does not have to be changed

Reply:
`BuildNumber` is only called on the templates we receive from the API. The empty build string from this `Entry` argument is never (tried to be) converted to an `int`.